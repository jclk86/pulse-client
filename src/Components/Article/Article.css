.container_article_image {
  width: 100%;
  margin: 30px auto;
}

.article_image {
  width: 100%;
  object-fit: cover;
  height: 300px;
}

.container_article {
  max-width: 800px;
  margin: 100px auto;
  padding-bottom: 50px;
  padding-right: 0;
}

.link_user_profile {
  text-decoration: none;
  color: rgb(51, 160, 255);
  font-weight: bold;
}

.link_user_profile:hover {
  text-decoration: underline;
}

.article_date,
.article_author {
  font-size: 16px;
}

.article_content {
  white-space: pre-line;
  text-align: justify;
  text-indent: 5%;
}

.Article_edit_post_btn {
  min-width: 200px;
  border-radius: 10px;
  font-size: 16px;
  border: none;
  background-color: rgb(90, 90, 90);
  padding: 10px;
  color: white;
  font-weight: bold;
  background-color: rgb(51, 160, 255);
}

.Article_edit_post_btn:hover {
  cursor: pointer;
}

.Article_edit_post_btn:focus {
  outline-width: 0;
}

@media only screen and (min-width: 568px) {
  .container_article_image {
    max-width: 800px;
  }
  .article_image {
    height: 400px;
    object-fit: cover;
  }
}
