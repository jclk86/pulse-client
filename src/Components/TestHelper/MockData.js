export const mockUser = {
  username: "Pop123",
  password: "Password123!",
  fullname: "Bob Smith",
  email: "BSmith@gmail.com"
};

export const mockArticle = {
  title: "test article",
  content: "test content",
  article_tag: "News",
  id: 1,
  date_created: "2019-10-14T21:49:20.570Z",
  author: {
    username: mockUser.username
  }
};

export const mockComment = {
  id: 1,
  content: "test comment content",
  date_created: "2019-10-14T21:49:20.570Z",
  article_id: 1,
  user: {
    username: mockArticle.author.username
  }
};
