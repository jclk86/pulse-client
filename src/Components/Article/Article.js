import React, { Component } from "react";
import { withRouter, NavLink } from "react-router-dom";
import ArticleContext from "../../Context/ArticleContext";
import TokenService from "../../Services/token-service";
import { Button } from "../Utils/Utils";
import "./Article.css";

class Article extends Component {
  static contextType = ArticleContext;

  render() {
    const { article } = this.props;
    const token = TokenService.readJwtToken();
    console.log(token.user_id);
    console.log(article.author.id);
    return (
      <div className="container_article">
        <div className="container_image">IMAGE SECTION</div>
        <h2>{article.title}</h2>
        <p>
          <span>By</span> {article.author.username}
        </p>
        <p>{article.content}</p>
        {token.user_id === article.author.id ? (
          <NavLink to={`/articles/${article.id}/edit_article`}>
            <Button type="button">Edit Post</Button>
          </NavLink>
        ) : (
          <div></div>
        )}
      </div>
    );
  }
}

export default withRouter(Article);
