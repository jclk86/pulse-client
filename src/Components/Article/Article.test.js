import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AuthApiService from "../../Services/auth-api-service";
import ArticleApiService from "../../Services/article-api-service";
import { mockUser, mockArticle } from "../TestHelper/MockData";
import Article from "./Article";

describe(`Article Component`, () => {
  beforeAll(() => {
    AuthApiService.postUser(mockUser).then(user => {
      AuthApiService.postLogin({
        username: user.username,
        password: user.password
      });
    });

    ArticleApiService.postArticle(mockArticle);
  });

  it("renders Article component without issues", () => {
    const wrapper = mount(
      <BrowserRouter>
        <Article article={mockArticle}></Article>
      </BrowserRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
