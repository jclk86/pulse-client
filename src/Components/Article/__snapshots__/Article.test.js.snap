// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Component renders Article component without issues 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Article)
      article={
        Object {
          "article_tag": "News",
          "author": Object {
            "username": "Pop123",
          },
          "content": "test content",
          "date_created": "2019-10-14T21:49:20.570Z",
          "id": 1,
          "title": "test article",
        }
      }
    >
      <Article
        article={
          Object {
            "article_tag": "News",
            "author": Object {
              "username": "Pop123",
            },
            "content": "test content",
            "date_created": "2019-10-14T21:49:20.570Z",
            "id": 1,
            "title": "test article",
          }
        }
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <div
          className="container_article"
        >
          <div
            className="container_image"
          >
            IMAGE SECTION
          </div>
          <h2>
            test article
          </h2>
          <p>
            Monday, October 14, 2019
          </p>
          <p>
            <span>
              By
            </span>
             
            Pop123
          </p>
          <p>
            test content
          </p>
          <NavLink
            to="/articles/1/edit_article"
          >
            <Link
              aria-current={null}
              to={
                Object {
                  "hash": "",
                  "pathname": "/articles/1/edit_article",
                  "search": "",
                  "state": null,
                }
              }
            >
              <LinkAnchor
                aria-current={null}
                href="/articles/1/edit_article"
                navigate={[Function]}
              >
                <a
                  aria-current={null}
                  href="/articles/1/edit_article"
                  onClick={[Function]}
                >
                  <Button
                    type="button"
                  >
                    <button
                      className="Button "
                      type="button"
                    >
                      Edit Post
                    </button>
                  </Button>
                </a>
              </LinkAnchor>
            </Link>
          </NavLink>
        </div>
      </Article>
    </withRouter(Article)>
  </Router>
</BrowserRouter>
`;
